plantcv/plantcv/morphology/fill_segments.py

```diff
@@ -55,21 +55,24 @@ def fill_segments(mask, objects, stem_objects=None, label="default"):
         outputs.add_observation(sample=label, variable='leaf_area', trait='segment area',
                                 method='plantcv.plantcv.morphology.fill_segments',
                                 scale='pixels', datatype=list,
-                                value=counts[1:].tolist(),
-                                label=(ids[1:]-1).tolist())
+                                value=counts[1:-1].tolist(),
+                                label=(ids[1:-1]-1).tolist())
         outputs.add_observation(sample=label, variable='stem_area', trait='segment area',
                                 method='plantcv.plantcv.morphology.fill_segments',
                                 scale='pixels', datatype=list,
-                                value=counts[1:].tolist(),
-                                label=(ids[1:]-1).tolist())
-
-    rgb_vals = color_palette(num=len(labels), saved=False)
-    filled_img = np.zeros((h, w, 3), dtype=np.uint8)
-    for l in labels:
-        for ch in range(3):
-            filled_img[:, :, ch][filled_mask == l] = rgb_vals[l - 1][ch]
-
+                                value=counts[-1].tolist(),
+                                label=(ids[-1]-1).tolist())
+
+    # rgb_vals = color_palette(num=len(labels), saved=False)
+    # filled_img = np.zeros((h, w, 3), dtype=np.uint8)
+    # for l in labels:
+    #     for ch in range(3):
+    #         filled_img[:, :, ch][filled_mask == l] = rgb_vals[l - 1][ch]
+
+    debug = params.debug
+    params.debug = None
+    filled_img = colorize_label_img(filled_mask)
+    params.debug = debug
     _debug(visual=filled_img, filename=os.path.join(params.debug_outdir, str(params.device) + "_filled_img.png"))
 
-
-    return filled_img
+    return filled_img, filled_mask
```

Comment:
Even though it's sort of a breaking change, I was thinking that rather than returning `filled_img` we would return `filled_mask` instead, only. But we would display `filled_img` in debug mode. And if the user wants to create the `filled_img` they can use the new `colorize_label_img` function. What's your opinion @DannieSheng and @HaleySchuhl?

Reply:
Did you mean that we generate the image and show `filled_img` in `plot` debug mode, but just not return it?

Reply:
Yeah, I'm proposing the code stays as you have it but that we only `return filled_mask`

Reply:
Well that makes sense to me. Users can view / save debug `filled_img` in `plot` / `print` mode, the `filled_img` itself might not be super useful in other calculations.